motionplan/ik/nloptInverseKinematics.go

```diff
@@ -86,6 +84,46 @@ func (ik *NloptIK) Solve(ctx context.Context,
 	//nolint: gosec
 	randSeed := rand.New(rand.NewSource(int64(rseed)))
 	var err error
+	mInput := &State{Frame: ik.model}
+
+	// Determine optimal jump values; start with default, and if gradient is zero, increase to 1 to try to avoid underflow.
+	jump := make([]float64, 0, len(seed))
+	seedTest := append(make([]referenceframe.Input, 0, len(seed)), seed...)
+	eePos, err := ik.model.Transform(seed)
+	if err != nil {
+		return err
+	}
+	mInput.Configuration = seed
+	mInput.Position = eePos
+	seedDist := solveMetric(mInput)
+	for i, testVal := range seedTest {
+		for jumpVal := defaultJump; jumpVal < 1; jumpVal *= 10 {
+			seedTest[i] = referenceframe.Input{testVal.Value + jumpVal}
+			if seedTest[i].Value > ik.upperBound[i] {
+				seedTest[i] = referenceframe.Input{testVal.Value - jumpVal}
+				if seedTest[i].Value < ik.lowerBound[i] {
+					jump = append(jump, defaultJump)
+					break
+				}
+			}
+			eePos, err = ik.model.Transform(seedTest)
+			if err != nil {
+				return err
+			}
+			mInput.Configuration = seed
```

Comment:
Why are we setting `mInput.Configuration` in a loop when the value of `seed` and `mInput` never change in the loop from lines 99 and 126.